backend/lib/services/members/MemberManager.js

```diff
@@ -234,6 +241,19 @@ class MemberManager {
     })
   }
 
+  handleRejectedPromises (results) {
+    const errors = _
+      .chain(results)
+      .filter(['status', 'rejected'])
+      .map('reason')
+      .filter(err => !isHttpError(err) || err.statusCode !== 404)
+      .value()
+
+    if (errors.length) {
+      throw new Error(errors)
+    }
+  }
```

Comment:
This should be a standalone function or a static method.
```suggestion
  static getFulfilledValues (results) {
    const errors = _
      .chain(results)
      .filter(['status', 'rejected'])
      .map('reason')
      .filter(err => !isHttpError(err) || err.statusCode !== 404)
      .value()

    if (errors.length === 1) {
      throw errors[0]
    } else if (errors.length > 1) {
      const message = errors.map(err => err.message).join('\n')
      throw new AggregateError(errors, message)
    }
    
    return _
      .chain(results)
      .filter(['status', 'fulfilled'])
      .map('value')
      .value()
  }
```